bigframes/ml/remote.py

```diff
@@ -0,0 +1,155 @@
+# Copyright 2023 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""BigFrames general remote models."""
+
+from __future__ import annotations
+
+from typing import Mapping, Optional, Union
+import warnings
+
+import bigframes
+from bigframes import clients
+from bigframes.ml import base, core, globals, utils
+import bigframes.pandas as bpd
+
+_SUPPORTED_DTYPES = (
+    "bool",
+    "string",
+    "int64",
+    "float64",
+    "array<bool>",
+    "array<string>",
+    "array<int64>",
+    "array<float64>",
+)
+
+_REMOTE_MODEL_STATUS = "remote_model_status"
+
+
+class VertexAIModel(base.BaseEstimator):
+    """Remote model from a Vertex AI https endpoint. User must specify https endpoint, input schema and output schema.
+    How to deploy a model in Vertex AI https://cloud.google.com/bigquery/docs/bigquery-ml-remote-model-tutorial#Deploy-Model-on-Vertex-AI.
+
+    Args:
+        endpoint (str):
+            Vertex AI https endpoint.
```

Comment:
Could you give an example of the endpoints? For example, I saw `"https://us-central1-aiplatform.googleapis.com/v1/projects/1084210331973/locations/us-central1/endpoints/3193318217619603456"` in your test. Might be give users some hints how to find the endpoints?

Reply:
User needs to deploy the model by themselves then they can find the endpoint, as well as the input/output schema. Such as https://screenshot.googleplex.com/49NX9PYEbBFNeMH